File: java_algo/target_number.java
class Solution {
    static int ret = 0;
    public int solution(int[] numbers, int target) {
        dfs(numbers, target, 0);
        return ret;
    }
    
    public void dfs(int [] numbers, int target, int pointer){
        if(pointer == numbers.length){
            int current_sum = 0;
            for(int j=0; j<numbers.length; j++){
                current_sum += numbers[j];
            }
            
            if(current_sum == target){
                ret ++;
            }
            return;
        }
        else{
            numbers[pointer]*=1;
            dfs(numbers,target,pointer+1);
            
            numbers[pointer]*=-1;
            dfs(numbers,target,pointer+1);
        }
    }
}

File: kakao/targetnumber.py
answer = 0


def solution(numbers, target, index=0):
    #answer = 0
    if index == len(numbers)-1:
        # then here do the calculation with sum after adding the values
        if sum(numbers) == target:
            answer += 1

        numbers[-1] = numbers[-1] * -1
        if sum(numbers) == target:
            answer += 1
    else:
        # first case
        solution(numbers, target, index+1)
        #second - case
        numbers[index] = numbers[index] * -1
        solution(numbers, target, index+1)
    print(target)
    return answer
